### DISCUSSION TITLE
Configuration for EFCore DbContext

### DISCUSSION THREAD

**Discussion by NickBuckland** [Finbuckle/Finbuckle.MultiTenant#662] [Q&A] [answered]
The link from the documentation for EFCoreStoreSamples does not work.

I'm struggle to figure out how to implement the tenant store using an EFCore Db which implements IdentityUser.

Can anyone provide an example?
Thanks

**Comment by AndrewTriesToCode** [comment, marked as answer]
Hi, I'm sorry the link is broken. [Here's a link to the last version that had it.](
https://github.com/Finbuckle/Finbuckle.MultiTenant/tree/v6.9.1/samples/ASP.NET%20Core%203/EFCoreStoreSample)

It's ASP.NET Core 3.0 but most of it is still applicable.

> I'm struggle to figure out how to implement the tenant store using an EFCore Db which implements IdentityUser.

Are you sure you want to do this? The multitenant store itself is not multitenant. It's the store of tenant data. I usually use a separate multitenant dbcontext for Identity so that I can have different sets of users/roles/etc for each tenant.

If you do want to use the same dbcontext make sure you have a way to initialize the dbcontext without an active tenant (because the store will be used before at tenant is known... to determine the tenant).

**Reply by NickBuckland** [reply]
Ah, got it thank you.
I was conflating Identity with Mutlitenant. Using a separate DbContext for each makes sense, thanks.

BTW thanks for the great work!
